ethcore/src/client/client.rs

```diff
@@ -204,6 +204,9 @@ pub struct Client {
 	/// Database pruning strategy to use for StateDB
 	pruning: journaldb::Algorithm,
 
+	/// Don't prune the state we're currently snapshotting
+	snapshotting_at: AtomicU64,
```

Comment:
Would it make more sense for it to be a `RwLock<Option<usize>>` ? Even though we would never take a snapshot at 0.

Reply:
I considered it. I think using an atomic here is faster (but not sure tbh) but it definitely reads better imo. It is written to once when a snapshot starts, and once again when it ends and in between only read. My thinking for omitting the `Option` here was that even if someone down the line would ask for as snapshot at block `0` due to a bug, nothing bad would happen, so the `Option` doesn't buy me much in terms of safety.

Reply:
Yeah I'm not to worried about a potential bug, it was mainly for readability